test/utils_test.rb

```diff
@@ -0,0 +1,171 @@
+require File.expand_path(File.join(File.dirname(__FILE__), "test_helper"))
+
+class RubySamlTest < Minitest::Test
+
+  describe "#format_cert" do
+    it "return empty string when the cert is an empty string" do
+      cert = ""
+      assert_equal cert, OneLogin::RubySaml::Utils.format_cert(cert, true)
+      assert_equal cert, OneLogin::RubySaml::Utils.format_cert(cert, false)
+    end
+
+    it "return formatted cert with head from cert with head" do
+      cert = ruby_saml_cert_text
+      formatted_cert = OneLogin::RubySaml::Utils.format_cert(cert, true)
+      assert cert.include? "-----BEGIN CERTIFICATE-----"
+      assert cert.include? "-----END CERTIFICATE-----"
+      assert_equal cert, formatted_cert
+      assert formatted_cert.include? "-----BEGIN CERTIFICATE-----"
+      assert formatted_cert.include? "-----END CERTIFICATE-----"
+    end
+
+    it "return formatted cert with head from cert without head" do
```

Comment:
you can do all of this cert checking logic with Ruby's `OpenSSL` module.  ie. `OpenSSL::X509::Certificate.new(cert_string)`

http://ruby-doc.org/stdlib-2.0/libdoc/openssl/rdoc/OpenSSL.html